### PR TITLE
Replay viewing fixes

### PR DESCRIPTION
**Pull request by MantasAD**
Made some fixes to improve replay viewing experience:
Fixed replay ghost's interpolated velocity calculation.
Fixed map finished panel not able to unhide cursor after a replay.
Made other small fixes.

### CODE REVIEW

File: mp/src/game/server/momentum/mom_replay_entity.cpp [hunk 1]
```diff
@@ -409,6 +393,22 @@ void CMomentumReplayGhostEntity::HandleGhostFirstPerson()
...
+{
+    if (m_pCurrentSpecPlayer->GetObserverMode() == OBS_MODE_IN_EYE)
+    {
+        HideGhost();
+    }
+    else
+    {
+        // we divide x angle (pitch) by 10 so the ghost doesn't look really stupid
+        angles.x /= GHOST_PITCH_REDUCTION_VALUE;
+        UnHideGhost();
```
Comment (Gocnak) [inline mp/src/game/server/momentum/mom_replay_entity.cpp]: Why are we setting ghost visibility in a method for setting the ghost's eye angles? Where it was before in `HandleGhostFirstPerson` is fine, moving and combining the functionality like this here could cause some unintended effects if things randomly set ghost angles in the future.

File: mp/src/game/server/momentum/mom_replay_entity.cpp [hunk 2]
```diff
@@ -70,12 +72,6 @@ void CMomentumReplayGhostEntity::Spawn()
     if (pPlayer)
     {
         SetGhostAppearance(pPlayer->m_playerAppearanceProps);
-        // now that we've set our appearance, the ghost should be visible again.
-        SetRenderMode(kRenderTransColor);
-        if (m_ghostAppearance.m_bGhostTrailEnable)
-        {
-            CreateTrail();
-        }
```
Comment (Gocnak) [inline mp/src/game/server/momentum/mom_replay_entity.cpp]: If this is removed, where does the ghost's trail get created now?

Comment (MantasAD) [reply mp/src/game/server/momentum/mom_replay_entity.cpp]: Ghost's trail is created after updating trail properties inside the base [class](https://github.com/momentum-mod/game/blob/2e2d5cd387fa5b5eb98e855b924fe34f1f728a87/mp/src/game/server/momentum/mom_ghost_base.cpp#L141)

File: mp/src/game/server/momentum/mom_replay_entity.cpp (hunk 1 above)
Comment (MantasAD) [reply mp/src/game/server/momentum/mom_replay_entity.cpp]: I wanted to extract that code into a function so I could use it at the very end of the replay when the ghost is stuck in place. Because before when the player changed his view to a third-person the ghost just wouldn't appear. Maybe I should rename the function to a more descriptive name.

File: mp/src/game/server/momentum/mom_replay_entity.cpp (hunk 2 above)
Comment (Gocnak) [reply mp/src/game/server/momentum/mom_replay_entity.cpp]: Ah okay. We may want to move the check for if it's enabled to inside CreateTrail so it doesn't create the trail if the appearance says not to.

File: mp/src/game/server/momentum/mom_replay_entity.cpp (hunk 1 above)
Comment (Gocnak) [reply mp/src/game/server/momentum/mom_replay_entity.cpp]: In that case I would pull the visibility out into its own `DetermineGhostVisibility()` and call that in the places you need to. It shouldn't embedded here.

File: mp/src/game/shared/momentum/mom_gamemovement.cpp [hunk 3]
```diff
@@ -930,7 +930,7 @@ void CMomentumGameMovement::FinishDuck(void)
     {
         Vector org = mv->GetAbsOrigin();
 
-        if (player->GetGroundEntity() != nullptr)
+        if (player->GetGroundEntity() != nullptr || m_pPlayer->m_CurrentSlideTrigger)
```
Comment (SlidyBat) [inline mp/src/game/shared/momentum/mom_gamemovement.cpp]: How is this related to the replay stuff?

Comment (Gocnak) [reply mp/src/game/shared/momentum/mom_gamemovement.cpp]: I let it slide after talking about it in the game code channel on Discord. They know for the future that small fixes like this should be in their own branch.
